### PR TITLE
Allow user to supply an arbitrary filter param to [GithubTag] and [GithubRelease] badges

### PR DESCRIPTION
**Pull request by chris48s**
This PR implements the solution discussed in #9173 and applies it to the GitHub Tag and GitHub Release badges, although we could apply the same approach to others in future.

We've talked through the reasons and use-cases for this, but this provides a fairly flexible solution which enables users to do things like `/github/v/tag/badges/shields?filter=%21server-%2A` in monorepos and other situations where repos contain a mix of different tags/releases.

Closes https://github.com/badges/shields/discussions/6253
Closes https://github.com/badges/shields/issues/6275
Closes https://github.com/badges/shields/pull/8205
Closes https://github.com/badges/shields/pull/9076
Closes https://github.com/badges/shields/issues/9080
Closes https://github.com/badges/shields/issues/9173

@calebcartwright  - Although it would be nice to get this merged, if you've got time to review stuff, I am more keen to get https://github.com/badges/shields/pull/9014 over the line tbh.

### CODE REVIEW

File: services/github/github-tag.spec.js [hunk 1]
```diff
@@ -53,4 +53,24 @@ describe('GithubTag', function () {
...
+
+  test(GithubTag.applyFilter, () => {
+    const tags = ['v1.1.0', 'v1.2.0', 'server-2022-01-01']
+    given({ tags, filter: undefined }).expect(tags)
+    given({ tags, filter: '' }).expect(tags)
+    given({ tags, filter: '*' }).expect(tags)
+    given({ tags, filter: '!*' }).expect([])
+    given({ tags, filter: 'foo' }).expect([])
+    given({ tags, filter: 'server-*' }).expect(['server-2022-01-01'])
+    given({ tags, filter: '!server-*' }).expect(['v1.1.0', 'v1.2.0'])
```
Comment (chris48s) [inline services/github/github-tag.spec.js:74]: I haven't really gone to town on testing different filters here. `matcher` has its own test suite.

I also haven't added any additional tests at the service layer. Just unit tests.

File: services/github/github-release.tester.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ t.create('Release (No releases)')
 
 t.create('Prerelease (No releases)')
   .get('/v/release/badges/daily-tests.json?include_prereleases')
-  .expectBadge({ label: 'release', message: 'no releases' })
+  .expectBadge({ label: 'release', message: 'no releases found' })
```
Comment (chris48s) [inline services/github/github-release.tester.js:35]: I've taken the opportunity here to align with the tag error message.

File: services/github/github-release.service.js [hunk 3]
```diff
@@ -85,6 +86,21 @@ class GithubRelease extends GithubAuthV3Service {
...
+        sort: 'date',
+        display_name: 'release',
+        filter: '*Open Beta',
+      },
+      staticPreview: this.render({
+        version: 'Release v6.0.2 - Open Beta',
+        sort: 'date',
+        isPrerelease: false,
+      }),
+      documentation: documentation + filterDocs,
```
Comment (chris48s) [inline services/github/github-release.service.js:102]: The move to OpenAPI in #9014 and follow-on PRs is going to make it way easier to document services that have lots of optional query params and provide documentation relevant to individual parameters.

File: services/github/github-common-release.js [hunk 4]
```diff
@@ -68,8 +69,39 @@ function getLatestRelease({ releases, sort, includePrereleases }) {
...
+  or more characters, and if the pattern starts with a <code>!</code>,
+  the whole pattern is negated.
+  </p>
+</div>`
+
+function applyFilter({ releases, filter, displayName }) {
+  if (!filter) {
+    return releases
+  }
+  if (displayName === 'tag') {
```
Comment (chris48s) [inline services/github/github-common-release.js:89]: Here we apply the filter to either the release names or tag names depending on which is going to get displayed on the badge

**Comment by chris48s** [comment]
The socket.dev issues are false positive. Turns out there are a lot of similarly named packages called lodash.something :)

File: services/github/github-tag.spec.js (hunk 1 above)
Comment (calebcartwright) [reply services/github/github-tag.spec.js:74]: I agree with the approach, we don't need to re-test the internals of a 3rd party library, just a couple of integration-like tests that give us confidence we're using that lib correctly.

Think you've fully cleared that threshold here :+1:

**Review by calebcartwright — APPROVED** [review]
lgtm, will defer to you on when to merge as I know there's some other PRs that you may want to juggle (e.g. ones touching the lockfile, ones implementing similar end results, etc.)

**Pull request merged**